designs/ui-module/README.md

```diff
@@ -0,0 +1,261 @@
+
+5. Dashboard layout sidebar  
+In dashboard layout sidebar, UI_SUBGROUP_I instance can be placed under the group.  
+Widget contained in UI_SUBGROUP_I instance is not displayed, only UI_SUBGROUP_I instance is displayed.  
+Pressing the [edit] button opens the edit screen for UI_SUBGROUP_I instance.  
+Layout Tabs & Lynks editor:  
+![layout-tab-link](./layout-tab-link.png)   
+*UI_SUBGROUP_I instance settings can be edited with the [edit] button on this screen.  
+To edit the layout of UI module(Template), open the UI_SUBGROUP_T template edit screen from the instance screen and press the [Edit Subgroup Layout] button to open the layout screen.
+
+### UI module function operation procedure
+
+The operation procedure when the user arranges the UI module on the dashboard is shown below.  
+
+1. Create SUBFLOW template  
+The user creates SUBFLOW template and places widget.
+
+2. Edit UI_SUBGROUP_T template  
+In step 1, Node-RED dashboard automatically creates UI_SUBGROUP_T template.  
+The user can edit the layout of UI module by opening the edit screen of UI_SUBGROUP_T template node from the node "info" tab.  
+
+3. Create SUBFLOW instance  
+Place SUBFLOW instance on tab or on SUBFLOW template.
+
+4. Edit UI_SUBGROUP_I instance  
+In step 3, Node-RED dashboard automatically generates UI_SUBGROUP_I instance.  
+The user opens the edit screen of UI_SUBGROUP_I instance node from the node "info" tab, etc., and specifies UI_GROUP of arbitrary Node-RED dashboard.  
+However, if you place it on SUBFLOW template, you do not need to specify UI_GROUP.
+
+5. Layout of dashboard  
+The user edits the layout of dashboard that contains the UI module.  
+UI module size is fixed.  
+
+6. SUBFLOW property support(optional)  
+If UI_GROUP is defined as the environment variable for the property definition of SUBFLOW, UI_GROUP of Node-RED dashboard can be specified in step 3.  
+In this case, the procedure for setting UI_GROUP in step 4 can be omitted. (See below)
+
+### UI module sharing
+
+UI module can be exported as SUBFLOW, which can be imported.  
+By setting the node ID of SUBFLOW template to the 'z' key, the SUBFLOW template that contains the "UI module configuration node" will be the export target.
+
+- UI_SUBGROUP_T template node  
+Set SUBFLOW template node to the 'z' key.
+
+- UI_SUBGROUP_I instance node  
+When arranging to SUBFLOW template: Set the ID of SUBFLOW template node to the 'z' key.  
+When arranging in TAB: 'z' key is not specified (set as Global Configlation Node).
+
+### Specification of UI_GROUP by SUBFLOW environment variable (optional function)
+
+By specifying UI_GROUP config node using the property editing of SUBFLOW, 
+you can specify the group that displays the UI component when placing SUBFLOW instance like normal widget.  
+
+Property name : `_UI_GROUP`
+
+UI_SUBGROUP_I instance node sets the value specified in the corresponding SUBFLOW instance _UI_GROUP property to the value specified in group.
+
```

Comment:
As above - why is this needed?

Reply:
By linking the value of _UI_GROUP to ui_subgroup_i, you can omit the setting of ui_group.